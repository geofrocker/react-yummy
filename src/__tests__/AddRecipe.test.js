import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { shallow, mount } from 'enzyme';
import AddRecipe from '../components/add_recipe';

it('renders without crashing', () => {
    const props = {
        history: {
            push: () => {}
        }
    }
    shallow(<AddRecipe {...props}/>);
});