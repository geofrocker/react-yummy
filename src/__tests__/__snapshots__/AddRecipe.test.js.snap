// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add recipe component renders the the add recipe form and submit data 1`] = `
<div
  className="AddRecipe"
>
  <h1>
    Add Recipe
  </h1>
  <div />
  <div
    className="jumbotron col-sm-8"
  >
    <form
      id="recipe-form"
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <input
          className="form-control"
          id="title"
          onChange={[Function]}
          placeholder="Title"
          required={true}
          type="text"
          value="Awesome recipe"
        />
      </div>
      <div
        className="form-group"
      >
        <select
          className="form-control"
          id="category"
          onChange={[Function]}
        >
          <option
            disabled={true}
            selected={true}
          >
            Select Category
          </option>
        </select>
      </div>
      <div
        className="form-group"
      >
        <input
          className="form-control"
          id="ingredients"
          onChange={[Function]}
          placeholder="Ingredients"
          required={true}
          type="text"
          value="one and two"
        />
      </div>
      <div
        className="form-group"
      >
        <textarea
          className="form-control"
          id="steps"
          onChange={[Function]}
          placeholder="Add your steps here"
          required={true}
          value="one and two"
        />
      </div>
      <select
        id="status"
        onChange={[Function]}
        required={true}
      >
        <option
          disabled={true}
          selected={true}
        >
          Select status
        </option>
        <option
          value="public"
        >
          Public
        </option>
        <option
          value="private"
        >
          Private
        </option>
      </select>
       
      <ess>
        * required
      </ess>
      <br />
      <br />
      <input
        className="btn btn-primary"
        type="submit"
        value="Submit"
      />
      Â 
      <a
        className="btn btn-success"
        href="/dashboard?tab=1"
      >
        Cancel
      </a>
    </form>
  </div>
</div>
`;
